Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,40 @@
+# Ref: http://wiki.bash-hackers.org/howto/mutex
+
+HOMEBREW_LOCK_DIR="$HOMEBREW_LIBRARY/lock.d"
+HOMWBREW_LOCK_PID_FILE="$HOMEBREW_LOCK_DIR/PID"
+
+lock() {
```

Comment:
OK, I see your point. However, such approach is kinda limit. For example, `ps` will show relative path if you run brew by `./bin/brew ...`.

Even more, it doesn't support lock inside brew invoking, which is what I needed when we start to auto install vendor ruby for old OS.